overlord/snapstate/snapstate.go

```diff
@@ -2217,36 +2217,124 @@ func LinkNewBaseOrKernel(st *state.State, name string) (*state.TaskSet, error) {
 	linkSnap := st.NewTask("link-snap", fmt.Sprintf(i18n.G("Make snap %q (%s) available to the system during remodel"), snapsup.InstanceName(), snapst.Current))
```

Comment:
I think it would be good to have unit tests for all these functions that look at the task set produced without running it using verifyUpdateTasks, that will require teaching that helper a few more options. The goal is that if we add tasks to the main update case we should have to take in consideration also these other functions. This could be done in a follow up.

Reply:
Addressed in #10943